app/controllers/submissions_controller.rb

```diff
@@ -1,6 +1,17 @@
 class SubmissionsController < ApplicationController
-  skip_before_action :authenticate_user!, only: [:new, :create, :thank_you]
-  layout 'dashboard', only: [:valid, :rated, :to_rate, :rejected, :results]
+  skip_before_action :authenticate_user!, only: [:confirm, :new, :create, :thank_you]
+  layout 'dashboard', only: [:valid, :all, :rated, :to_rate, :rejected, :results]
+
+  def confirm
+    token = params[:confirmation_token]
+    submission = Submission.find_by(confirmation_token: token)
+    if submission.has_expired?
+      render text: "Time for confirmation expired!"
+    else
+      submission.confirmed!
```

Comment:
It's method from rails enum for confirmation_status. Do you recommend just wrapping this method

`def confirm!
    self.confirmed!
  end`

or maybe setting confirmation_status explicitly:

`def confirm!
    self.confirmation_status = 'confirmed'
    save
  end`